test/functional/legacy/062_tab_pages_spec.lua

```diff
@@ -0,0 +1,330 @@
+      unlet nr
+    ]])
+
+    -- Open three tab pages and use ":tabdo".
+    source([[
+      0tabnew
+      1tabnew
+      $tabnew
+      tabdo call append(line('$'), 'this is tab page ' . tabpagenr())
+      tabclose! 2
+      tabrewind
+      let line1 = getline('$')
+      undo
+      q
+      tablast
+      let line2 = getline('$')
+      q!
+      call append(line('$'), line1)
+      call append(line('$'), line2)
+      unlet line1 line2
+    ]])
+
+    -- Test for settabvar() and gettabvar() functions. Open a new tab page and
+    -- set 3 variables to a number, string and a list. Verify that the
+    -- variables are correctly set.
+    source([=[
+      tabnew
+      tabfirst
+      call settabvar(2, 'val_num', 100)
+      call settabvar(2, 'val_str', 'SetTabVar test')
+      call settabvar(2, 'val_list', ['red', 'blue', 'green'])
+
+      let test_status = 'gettabvar: fail'
+      if  gettabvar(2, 'val_num') == 100 && 
+	\ gettabvar(2, 'val_str') == 'SetTabVar test' && 
+	\ gettabvar(2, 'val_list') == ['red', 'blue', 'green']
+          let test_status = 'gettabvar: pass'
+      endif
+      call append(line('$'), test_status)
+
+      tabnext 2
+      let test_status = 'settabvar: fail'
+      if  t:val_num == 100 && 
+	\ t:val_str == 'SetTabVar test' &&
+	\ t:val_list == ['red', 'blue', 'green']
+         let test_status = 'settabvar: pass'
+      endif
+      tabclose
+      call append(line('$'), test_status)
+    ]=])
+
+    -- Test some drop features 
+    -- TODO: which of these conditions in the first if are necessary?
+    -- Currently we override them all with " || has('nvim')".
+    source([[
+      if has('gui') || has('clientserver') || has('nvim')
+	" Test for ":tab drop exist-file" to keep current window.
+	sp test1
+	tab drop test1
+	let test_status = 'tab drop 1: fail'
+	if tabpagenr('$') == 1 && winnr('$') == 2 && winnr() == 1
+	    let test_status = 'tab drop 1: pass'
+	endif
+	close
+	call append(line('$'), test_status)
+
+	" Test for ":tab drop new-file" to keep current window of tabpage 1.
+	split
+	tab drop newfile
+	let test_status = 'tab drop 2: fail'
+	if  tabpagenr('$') == 2 && 
+	  \ tabpagewinnr(1, '$') == 2 &&
+	  \ tabpagewinnr(1) == 1
+	    let test_status = 'tab drop 2: pass'
+	endif
+	tabclose
+	q
+	call append(line('$'), test_status)
+
+	" Test for ":tab drop multi-opend-file" to keep current tabpage and
+	" window.
+	new test1
+	tabnew
+	new test1
+	tab drop test1
+	let test_status = 'tab drop 3: fail'
+	if  tabpagenr() == 2 && 
+	  \ tabpagewinnr(2, '$') == 2 &&
+	  \ tabpagewinnr(2) == 1
+	    let test_status = 'tab drop 3: pass'
+	endif
+	tabclose
+	q
+	call append(line('$'), test_status)
+      else
+      " :drop not supported.
+	call append(line('$'), 'tab drop 1: pass')
+	call append(line('$'), 'tab drop 2: pass')
+	call append(line('$'), 'tab drop 3: pass')
+      endif
+    ]])
+-------
+
+    execute('for i in range(9) | tabnew | endfor')
+    feed('1gt<cr>')
```

Comment:
You where right in this case (in general enter in normal mode command moves to the next line).